doozerlib/cli/get_nightlies.py

```diff
@@ -1,97 +1,186 @@
+from typing import List, Dict, Optional, Set, Tuple
+import asyncio
 import click
+import hashlib
 import json
 import sys
 from urllib import request
-from doozerlib.cli import cli
-from doozerlib import constants, util, exectools
-from doozerlib.rhcos import get_primary_container_name
+from doozerlib.cli import cli, click_coroutine
+from doozerlib.model import Model
+from doozerlib import constants, util, exectools, logutil
+from doozerlib.rhcos import RHCOSBuildInspector
 
+logger = logutil.getLogger(__name__)
 
-@cli.command("get-nightlies", short_help="Get sets of Accepted nightlies. A set contains nightly for each arch, "
-                                         "determined by closest timestamps")
-@click.option("--limit", default=1, metavar='NUM', help="Number of sets of passing nightlies to print")
+
+@cli.command("get-nightlies", short_help="Determine set(s) of accepted nightlies with matching contents for all architectures")
+@click.option("--matching", metavar="NIGHTLY_NAME", multiple=True, help="Only report nightlies with the same content as named nightly")
+@click.option("--allow-pending", is_flag=True, help="Include nightlies that have not completed tests")
+@click.option("--allow-rejected", is_flag=True, help="Include nightlies that have failed tests")
+@click.option("--exclude-arch", metavar="ARCH", multiple=True, help="Exclude arch(es) normally included in this version (multi,aarch64,...)")
+@click.option("--limit", default=1, metavar='NUM', help="Number of sets of nightlies to print")
 @click.option("--rhcos", is_flag=True, help="Print rhcos build id with nightlies")
```

Comment:
"doozer --quiet -g openshift-4.10 get-nightlies --rhcos" is not printing the build ids right now..

Reply:
i tried making it into `--details` instead

Reply:
Tried testing it, but I'm getting too many open files error 
```
Traceback (most recent call last):
  File "/home/sid/.local/bin/doozer", line 11, in <module>
    load_entry_point('rh-doozer', 'console_scripts', 'doozer')()
  File "/home/sid/Projects/art/doozer/doozerlib/cli/__main__.py", line 2234, in main
    cli(obj={})
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/decorators.py", line 38, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/sid/Projects/art/doozer/doozerlib/cli/__init__.py", line 167, in wrapper
    return loop.run_until_complete(f(*args, **kwargs))
  File "/usr/lib64/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/home/sid/Projects/art/doozer/doozerlib/cli/get_nightlies.py", line 107, in get_nightlies
    await eq_set.populate_deeper_equivalence(runtime)
  File "/home/sid/Projects/art/doozer/doozerlib/cli/get_nightlies.py", line 453, in populate_deeper_equivalence
    await asyncio.gather(*(
  File "/home/sid/Projects/art/doozer/doozerlib/cli/get_nightlies.py", line 343, in populate_deeper_equivalence
    await asyncio.gather(*(
  File "/home/sid/Projects/art/doozer/doozerlib/cli/get_nightlies.py", line 334, in retrieve_image_info_async
    image_json_str, _ = await exectools.cmd_assert_async(
  File "/home/sid/Projects/art/doozer/doozerlib/exectools.py", line 374, in cmd_assert_async
    result, out, err = await cmd_gather_async(cmd_list, text_mode=text_mode, cwd=cwd, set_env=set_env, strip=strip, log_stdout=log_stdout, log_stderr=log_stderr)
  File "/home/sid/Projects/art/doozer/doozerlib/exectools.py", line 309, in cmd_gather_async
    proc = await asyncio.create_subprocess_exec(
  File "/usr/lib64/python3.10/asyncio/subprocess.py", line 218, in create_subprocess_exec
    transport, protocol = await loop.subprocess_exec(
  File "/usr/lib64/python3.10/asyncio/base_events.py", line 1667, in subprocess_exec
    transport = await self._make_subprocess_transport(
  File "/usr/lib64/python3.10/asyncio/unix_events.py", line 207, in _make_subprocess_transport
    transp = _UnixSubprocessTransport(self, protocol, args, shell,
  File "/usr/lib64/python3.10/asyncio/base_subprocess.py", line 36, in __init__
    self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,
  File "/usr/lib64/python3.10/asyncio/unix_events.py", line 799, in _start
    self._proc = subprocess.Popen(
  File "/usr/lib64/python3.10/subprocess.py", line 969, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib64/python3.10/subprocess.py", line 1744, in _execute_child
    errpipe_read, errpipe_write = os.pipe()
OSError: [Errno 24] Too many open files
```

Reply:
Huh so that ^ only happens when I run the command in "Tilix" but not in vscode/pycharm terminal. The output looks good otherwise

Reply:
interesting. i never saw that, but i have:
```
$ ulimit -n
524288
```
... so it would be hard to run out of open files. i wonder if there's a way to limit the concurrency though... this is the point where it's running `oc image info` against all the images in all the nightlies at once...